client/src/app/pages/applications/components/application-detail-drawer/review-fields.tsx

```diff
@@ -0,0 +1,174 @@
+  application,
+}) => {
+  const { archetypes } = useFetchArchetypes();
+  const { reviews } = useFetchReviews();
+  const { t } = useTranslation();
+
+  const { review: appReview } = useFetchReviewById(application?.review?.id);
+
+  const notYetReviewed = (
+    <EmptyTextMessage message={t("terms.notYetReviewed")} />
+  );
+
+  const reviewedArchetypes = application?.archetypes
+    ?.map((archetypeRef) => {
+      return archetypes.find((archetype) => archetype.id === archetypeRef.id);
+    })
+    .filter(Boolean);
+
+  const hasReviewedArchetype = reviewedArchetypes?.some(
+    (archetype) => !!archetype?.review
+  );
+
+  const matchedArchetypeReviews: Review[] = (reviewedArchetypes || [])
+    .map((reviewedArchetype) => {
+      return reviews.find(
+        (review) => review.id === reviewedArchetype?.review?.id
+      );
+    })
+    .filter(Boolean);
+
+  const groupedReviewList = [
+    {
+      review: appReview,
+      name: appReview?.application?.name,
+      type: "App",
+    },
+    ...matchedArchetypeReviews.map((archetypeReview) => ({
+      review: archetypeReview,
+      name: archetypeReview?.archetype?.name,
+      type: "Archetype",
+    })),
+  ].filter((item) => item.review?.proposedAction);
+
+  return (
+    <>
+      <TextContent className={spacing.mtLg}>
+        <Title headingLevel="h3" size="md">
+          {t("terms.review")}
+        </Title>
+      </TextContent>
+      <DescriptionListGroup>
+        <DescriptionListTerm>{t("terms.proposedAction")}</DescriptionListTerm>
+        <DescriptionListDescription cy-data="proposed-action">
+          {groupedReviewList.length > 0
+            ? groupedReviewList.map((item, index) => (
+                <Label key={index} className={spacing.mbSm}>
+                  {item.type === "App"
+                    ? `App - ${item.name}`
+                    : item.type === "Archetype"
+                    ? `Archetype - ${item.name}`
+                    : "Unknown"}
+                  -{" "}
+                  {item.review?.proposedAction &&
+                  PROPOSED_ACTION_LIST[item.review?.proposedAction]
+                    ? t(PROPOSED_ACTION_LIST[item.review.proposedAction].i18Key)
+                    : "Unknown"}
+                </Label>
+              ))
+            : notYetReviewed}
+        </DescriptionListDescription>
+      </DescriptionListGroup>
+      <DescriptionListGroup>
+        <DescriptionListTerm>{t("terms.effortEstimate")}</DescriptionListTerm>
+        <DescriptionListDescription cy-data="effort-estimate">
+          {groupedReviewList.length > 0
+            ? groupedReviewList.map((item, index) => (
+                <Label key={index} className={spacing.mbSm}>
+                  {item.type === "App"
+                    ? `App - ${item.name}`
+                    : item.type === "Archetype"
+                    ? `Archetype - ${item.name}`
+                    : "Unknown"}
```

Comment:
This seems to be repeated a few times, it is a good candidate to push directly into a `groupedReviewList` item prop...maybe `displayName` or something similar